### PR TITLE
refine order by primary key

### PR DESCRIPTION
**Pull request by jiyfhust**
### What problem does this PR solve? 

https://github.com/pingcap/dumpling/issues/187

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### CODE REVIEW

**Comment by jiyfhust** [comment]
@lichunzhu @kennytm

**Review by lichunzhu — COMMENTED** [review]
LGTM

File: v4/export/sql.go [hunk 1]
```diff
@@ -315,6 +318,26 @@ func GetColumnTypes(db *sql.Conn, fields, database, table string) ([]*sql.Column
...
+	var cols []string
+	var col string
+	for rows.Next() {
+		err = rows.Scan(&col)
+		if err != nil {
+			return nil, errors.Annotatef(err, "sql: %s", priKeyColsQuery)
+		}
+		cols = append(cols, col)
+	}
+	return cols, nil
```
Comment (kennytm) [inline v4/export/sql.go:338]: also check `rows.Err()` before returning.

**Review by kennytm — COMMENTED** [review]
rest LGTM

File: v4/export/sql.go (hunk 1 above)
Comment (jiyfhust) [reply v4/export/sql.go:341]: done. Please review it agein.

**Comment by jiyfhust** [comment]
hi, @lichunzhu Please have a check again.

**Review by lichunzhu — APPROVED** [review]
LGTM

**Comment by lichunzhu** [comment]
thanks for the contribution!

**Pull request merged**